src/operation/equals.hpp

```diff
@@ -0,0 +1,72 @@
+// Unless explicitly stated otherwise all files in this repository are
+// dual-licensed under the Apache-2.0 License or BSD-3-Clause License.
+//
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021 Datadog, Inc.
+
+#pragma once
+
+#include <clock.hpp>
+#include <operation/base.hpp>
+#include <string_view>
+#include <unordered_map>
+#include <utils.hpp>
+
+namespace ddwaf::operation {
+
+template <typename T> class equals : public base_impl<equals<T>> {
+public:
+    using rule_data_type = std::vector<std::pair<std::string_view, uint64_t>>;
+
+    explicit equals(T expected) : expected_(std::move(expected)) {}
+    ~equals() override = default;
+    equals(const equals &) = default;
+    equals(equals &&) noexcept = default;
+    equals &operator=(const equals &) = default;
+    equals &operator=(equals &&) noexcept = default;
+
+protected:
+    static constexpr std::string_view to_string_impl() { return ""; }
+    static constexpr std::string_view name_impl() { return "equals"; }
+
+    template <typename U = T, std::enable_if_t<std::is_same<U, int64_t>::value, bool> = true>
```

Comment:
could be a bit simplified with

```c++
    template <typename U = T, typename = std::enable_if_t<std::is_same_v<U, int64_t>>>
```

or since c++20 is enabled...

```c++
    static constexpr DDWAF_OBJ_TYPE supported_type_impl() requires std::is_same_v<T, int64_t>
    {
        return DDWAF_OBJ_SIGNED;
    }
```

but still my preference would be a single definition and `if constexpr`.

Reply:
Ah... the burden of choice!

Reply:
I'll use `if constexpr` since it's more compact and I've used it in other places as well in this PR.